docs/usage.md

```diff
@@ -4,61 +4,107 @@ This is lazy dog HOWTO to using CodeChecker analysis.
+- [Unique Report Identifier (RI) <a name="unique-report-identifier"></a>](#unique-report-identifier-ri-)
+  - [Listing and Counting Reports <a name="listing-reports"></a>](#listing-and-counting-reports-)
+    - [How reports are counted? <a name="how-report-are-counted"></a>](#how-reports-are-counted-)
+  - [Report Uniqueing <a name="report-uniqueing"></a>](#report-uniqueing-)
+  - [How diffs between runs are calculated? <a name="diffs-between-runs"></a>](#how-diffs-between-runs-are-calculated-)
+
+## Preface<a name="preface"></a>
+The purpose of this document is to make the developer's first steps easier in
+usage of CodeChecker. The document is a mini course with simple examlpes that
+guides the developer how he/she can apply CodeChecker in his/her daily routine
+to make his/her code more roubust.
+
+There is a simple example program in the repository of CodeChecker what do
+almost nothing, but it can be used to demonstrate the abilities of
+CodeChecker. In this chapter that example program will be used to show
+CodeChecker usage.
+
+Example program placed in the [docs/examples](examples) directory. After
+repository of CodeChecker cloned step in it and follow statements in this
+document.
+
+There are some prerequisite to successfully take this example. First,
+CodeChecker should be [installed](README.md/#install-guide).
+Python 3 (> 3.6) should be installed on the host. Components of clang and llvm
+should be reached by PATH (clang, clang++, clang-tidy, etc.). (On debian based
+systems update-alternatives is your friend.) To compile example program `g++`
+and `make` is necessary to installed on host.
+
+## Step 1: Integrate CodeChecker into your build system<a name="step-1"></a>
+CodeChecker analyzes sources and dependencies that are built by your
+build system.
+
+### Step in the docs/examples directory<a name="step-in"></a>
+```sh
+cd <repo root dir>/docs/examples
+```
+
+### Clean the workspace<a name="clean-workspace"></a>
 ```sh
 make clean
 ```
-3. Log your build:
+
+### Log your build<a name="log-your-build"></a>
+Logging means that during the whole compilation process CodeChecker catches
+compiler calls and saves commands in a *compilation database* file.
+([Specification is here](https://clang.llvm.org/docs/JSONCompilationDatabase.html).)
+This compilation database is an input of the next analysis step.
+
 ```sh
-CodeChecker log -b "make" -o compilation.json
+CodeChecker log --build "make" --output ./compile_commands.json
 ```
-4. Check the contents of compilation.json. If everything goes well it should
-contain the `gcc` calls.
+
+### Check the contents of compile_commands.json file<a name="check-compile-commands"></a>
+If everything goes well it should contain the C compiler calls (`gcc`, `clang`,
+etc.).
+
 ```sh
-cat ./compilation.json
+cat ./compile_commands.json
 ```
 
-**What to do if the `compilation.json` is empty?**
+**What to do if the `compile_commands.json` is empty?**
```

Comment:
It would be good to mention here that a debug log file is generated beside the compilation_database.json (codechecker.logger.debug) and if the compilation database is empty maybe it will contain much information about this.

Reply:
Done.